### PR TITLE
feat(mini-nn): add baseline neural network for PlayerStatistics class…

### PR DESCRIPTION
**Pull request by LalitYemireddy**
### What I changed

* Added new file: `src/mini_nn/lalit_model.py`
* Built a small PyTorch neural network (`MiniNet`) for binary classification
* Model predicts whether a player scores **20+ points** in a game
* Uses data from `PlayerStatistics.csv`
* Removed scoring-related columns (FG made, 3PT made, FT made, etc.) to prevent target leakage
* Includes:

  * Train/validation/test split
  * Feature scaling (`StandardScaler`)
  * Early stopping
  * F1 score and AUC evaluation

---

### Why I changed it

This introduces a **baseline neural network model** to evaluate learning performance on NBA player game statistics. It helps:

* Validate the data pipeline end-to-end
* Establish a baseline before using more complex models
* Detect and prevent feature leakage
* Provide a foundation for future feature engineering

---

### How did I test it

Ran locally inside the project virtual environment:

python src/mini_nn/lalit_model.py

**Dataset**

* Rows: **1,655,736**
* Features used: **12**
* Positive rate (20+ pts): **12.9%**

**Final Test Results**
Test Accuracy: 0.7249
Test F1:       0.4319
Test AUC:      0.8374

Model trains successfully and produces realistic (non-leaky) performance.

---

### Related Issues

N/A – baseline model addition

---

### PR Title (Conventional Commit)

feat(mini-nn): add baseline neural network for PlayerStatistics classification

### CODE REVIEW

**Review by JonathanPLev — APPROVED** [review]
looks great, no questions. good job!
